core/src/main/java/com/datastax/oss/driver/api/core/metadata/schema/FunctionSignature.java

```diff
@@ -30,38 +32,45 @@
  */
 @Immutable
 public class FunctionSignature {
-  private final CqlIdentifier name;
-  private final List<DataType> parameterTypes;
+  @NonNull private final CqlIdentifier name;
+  @NonNull private final List<DataType> parameterTypes;
 
-  public FunctionSignature(CqlIdentifier name, Iterable<DataType> parameterTypes) {
+  public FunctionSignature(
+      @NonNull CqlIdentifier name, @NonNull Iterable<DataType> parameterTypes) {
     this.name = name;
     this.parameterTypes = ImmutableList.copyOf(parameterTypes);
   }
 
-  public FunctionSignature(CqlIdentifier name, DataType... parameterTypes) {
-    this(name, ImmutableList.<DataType>builder().add(parameterTypes).build());
+  public FunctionSignature(@NonNull CqlIdentifier name, @Nullable DataType... parameterTypes) {
+    this(
+        name,
+        parameterTypes == null
+            ? ImmutableList.of()
+            : ImmutableList.<DataType>builder().add(parameterTypes).build());
```

Comment:
Small glitch here but this is more an issue in the initial implementation: if the constructor is called with no varargs, `parameterTypes` is empty, not null. I think it's impossible to get null, I'll fix the method and mark it as non nullable.

Reply:
The Spotbugs annotations do not declare `@Target(ElementType.TYPE_USE)`, which is needed for the "double annotation".
I'll just annotate element nullability and document array nullability in the javadocs.

Reply:
IDEA validates it the other way around compared to what's explained in the stackoverflow answer: `void foo(@NonNull Object... objects)` means that the vararg array itself can't be null. There is no way to indicate if array elements are nullable, so that has to go in the javadoc.